crates/e2e-testing/src/utils.rs

```diff
@@ -1,63 +1,56 @@
 use anyhow::{anyhow, Result};
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
+use std::pin::Pin;
 use std::str;
 use std::{
     collections::HashMap,
-    ffi::OsStr,
     net::{Ipv4Addr, SocketAddrV4, TcpListener},
     process::{self, Command, Output},
     time::Duration,
 };
 
-use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::io::{AsyncBufRead, AsyncBufReadExt};
 use tokio::process::Command as TokioCommand;
-use tokio::process::{ChildStderr, ChildStdout};
 use tokio::time::timeout;
 use tokio::{net::TcpStream, time::sleep};
 
-/// run the command and returns the output
-pub fn run<S: Into<String> + AsRef<OsStr>>(
-    args: Vec<S>,
-    dir: Option<S>,
+/// Run the command and returns the output
+pub fn run<S: AsRef<str>>(
+    args: &[S],
+    dir: Option<&Path>,
     envs: Option<HashMap<&str, &str>>,
 ) -> Result<Output> {
     let mut cmd = Command::new(get_os_process());
     cmd.stdout(process::Stdio::piped());
     cmd.stderr(process::Stdio::piped());
 
     if let Some(dir) = dir {
-        cmd.current_dir(dir.into());
+        cmd.current_dir(dir);
     };
 
     cmd.arg("-c");
-    cmd.arg(
-        args.into_iter()
-            .map(Into::into)
-            .collect::<Vec<String>>()
-            .join(" "),
-    );
+    cmd.arg(args.iter().map(AsRef::as_ref).collect::<Vec<_>>().join(" "));
     if let Some(envs) = envs {
         for (k, v) in envs {
             cmd.env(k, v);
         }
     }
 
-    let output = cmd.output()?;
-    let code = output.status.code().expect("should have status code");
+    println!("Running command: {cmd:?}");
```

Comment:
do we want to move this to debug instead of printing always?

Reply:
I originally did this so that it was clear if the tests panicked due to the command failing what the command was (since the command itself is no longer printed in the panic message). I'm not against moving to `debug` but this might make it less obvious what went wrong when something did. Perhaps we leave it for now, and see how annoying it is before changing it?